testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/logout/OidcLogoutBaseTest.java

```diff
@@ -0,0 +1,521 @@
+import org.wildfly.security.jose.util.JsonSerialization;
+import org.wildfly.test.integration.elytron.oidc.client.KeycloakConfiguration;
+import org.wildfly.test.integration.elytron.oidc.client.KeycloakContainer;
+
+import io.restassured.RestAssured;
+
+/**
+ * Tests for the OpenID Connect logout types.
+ */
+public abstract class OidcLogoutBaseTest {
+
+    private static HttpClient httpClient;
+
+    @Before
+    public void createHttpClient() {
+        CookieStore store = new BasicCookieStore();
+        httpClient = TestHttpClientUtils.promiscuousCookieHttpClientBuilder()
+                .setDefaultCookieStore(store)
+                .setRedirectStrategy(new LaxRedirectStrategy())
+                .build();
+    }
+
+    @BeforeClass
+    public static void checkDockerAvailability() {
+        assumeTrue("Docker isn't available, OIDC tests will be skipped", AssumeTestGroupUtil.isDockerAvailable());
+    }
+
+    protected static OidcClientConfiguration config = new OidcClientConfiguration();
+
+    public static final String CLIENT_SECRET = "longerclientsecretthatisstleast256bitslong";
+    public static final String WEB_XML = "web.xml";
+    public static KeycloakContainer KEYCLOAK_CONTAINER;
+    public static final String TEST_REALM = "WildFly";
+    private static final String KEYCLOAK_USERNAME = "username";
+    private static final String KEYCLOAK_PASSWORD = "password";
+    public static final int CLIENT_PORT = TestSuiteEnvironment.getHttpPort();
+    // This name enables the Docker container (running keycloak) to make the
+    // local machine's host accessible to keycloak.
+    public static final String HOST_TESTCONTAINERS_INTERNAL = "host.testcontainers.internal";
+    public static final String RP_INITIATED_LOGOUT_APP = "RpInitiatedLogoutApp";
+    public static final String FRONT_CHANNEL_LOGOUT_APP = "FrontChannelLogoutApp";
+    public static final String BACK_CHANNEL_LOGOUT_APP = "BackChannelLogoutApp";
+    public static final String POST_LOGOUT_APP = "PostLogoutApp";
+
+    private final Stability desiredStability;
+
+    public OidcLogoutBaseTest(Stability desiredStability) {
+        this.desiredStability = desiredStability;
+    }
+
+    @Test
+    @OperateOnDeployment(RP_INITIATED_LOGOUT_APP)
+    public void testRpInitiatedLogout() throws Exception {
+
+        loginToApp(RP_INITIATED_LOGOUT_APP);
+        assertUserLoggedIn(RP_INITIATED_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(RP_INITIATED_LOGOUT_APP, "You are logged out");
+        assertUserLoggedOut(RP_INITIATED_LOGOUT_APP, "GOOD");
+    }
+
+    @Test
+    @OperateOnDeployment(FRONT_CHANNEL_LOGOUT_APP)
+    public void testFrontChannelLogout() throws Exception {
+
+        loginToApp(FRONT_CHANNEL_LOGOUT_APP);
+        assertUserLoggedIn(FRONT_CHANNEL_LOGOUT_APP, "GOOD");
+        logoutOfKeycloak(FRONT_CHANNEL_LOGOUT_APP, "You are logging out from following apps");
+        assertUserLoggedOut(FRONT_CHANNEL_LOGOUT_APP, "GOOD");
```

Comment:
Just to check, why is the expected text "GOOD" here? Wouldn't that indicate that the user is still logged in?